unlock-app/src/pages/_app.js

```diff
@@ -5,80 +5,32 @@ import { WalletService, Web3Service } from '@unlock-protocol/unlock-js'
 import TagManager from 'react-gtm-module'
 import configure from '../config'
 
-import GlobalStyle from '../theme/globalStyle'
 import { ConfigContext } from '../utils/withConfig'
 import { WedlockServiceContext } from '../contexts/WedlocksContext'
 import WedlockService from '../services/wedlockService'
 import ProviderContext from '../contexts/ProviderContext'
-import Authenticate from '../components/interface/Authenticate'
+import GlobalWrapper from '../components/interface/GlobalWrapper'
 import '../index.css'
 
 const config = configure()
 const wedlockService = new WedlockService(config.services.wedlocks.host)
-
 class UnlockApp extends App {
   constructor(props, context) {
     super(props, context)
-    this.state = {
-      provider: null,
-    }
 
```

Comment:
How about converting this into a functional component and making the file typescript?

Reply:
That's a good point!

Reply:
Actually not sure if this is possible since `UnlockApp` extends Next's `App`?